source/change-streams/tests/README.rst

```diff
@@ -0,0 +1,156 @@
+  - ``maxServerVersion``: Reserved for later use
+  - ``failPoint``: Reserved for later use
+  - ``target``: The entity to run a changeStream on. Valid values are:
+  
+    - ``collection``: Watch changes on collection ``database_name.collection_name``
+    - ``db``: Watch changes on database ``database_name``
+    - ``client``: Watch changes on entire clusters
+  - ``topology``: An array of server topologies to run the test against
+    Valid topologies are ``single``, ``replicaset``, and ``sharded``.
+  - ``changeStreamPipeline``: An array of additional aggregation pipeline stages to add to the change stream
+  - ``changeStreamOptions``: Additional options to add to the changeStream
+  - ``operations``: Array of documents, each describing an operation. Each document has the following fields:
+    - ``database``: Database to run the operation against
+    - ``collection``: Collection to run the operation against
+    - ``commandName``: Name of the command to run
+    - ``arguments``: Array of arguments for the command.
+
+  - ``expectations``: Optional list of command-started events
+  - ``result``: Document with ONE of the following fields:
+
+    - ``error``: Describes an error received during the test
+    - ``success``: An array of documents expected to be received from the changeStream
+
+Spec Test Match Function
+========================
+
+The definition of MATCH or MATCHES in the Spec Test Runner is as follows:
+
+- MATCH takes two values, ``expected`` and ``actual``
+- Notation is "Assert [actual] MATCHES [expected]
+- Assertion passes if ``expected`` is a subset of ``actual``, with the values ``42`` and ``"42"`` acting as placeholders for "any value"
+
+Pseudocode Implementation of ``actual`` MATCHES ``expected``:
+
+- If ``expected`` is ``"42"`` or ``42``
+
+  - Assert that ``actual`` exists (is not ``null`` or ``undefined``)
+
+- Else
+
+  - Assert that ``actual`` is of the same JSON type as ``expected``
+  - If ``expected`` is a ``number``, ``string``, ``boolean`` or ``null``
+
+    - Assert that ``expected`` equals ``actual``
+
+  - Else if ``expected`` is an ``array``
+
+    - for every ``idx``/``value`` in ``expected``
+
+      - Assert that ``value`` MATCHES ``actual[idx]``
+
+  - Else
+
+    - for every ``key``/``value`` in ``expected``
+
+      - Assert that ``value`` MATCHES ``actual``
+
+Spec Test Runner
+================
+
+Before running the tests
+
+- Create a MongoDB server topology.
+- Create a MongoClient ``globalClient``, and connect to the server
+
+For each YAML file, for each element in ``tests``:
+
+- If ``topology`` does not include the topology of the server instance(s), skip this test.
+- Use ``globalClient`` to
+
+  - Drop the database ``database_name``
+  - Create the database ``database_name`` and the collection ``database_name.collection_name``
+
+- Create a new MongoClient ``client``, with APM enabled
```

Comment:
Is APM actually something that is enabled when a client is constructed? IIRC, the [APM spec](https://github.com/mongodb/specifications/blob/master/source/command-monitoring/command-monitoring.rst) doesn't talk about registering APM individual clients but it does have an example for registering a global event subscriber. I suppose this may vary by driver.

Would it make sense to simply have "Register APM listeners for `client`" as its own bullet?

Reply:
I wanted to explicitly make clear that we should only be recording APM events from `client`, and not from the `globalClient`